pkg/backend/filestate/state.go

```diff
@@ -441,3 +441,34 @@ func (b *localBackend) addToHistory(name tokens.QName, update backend.UpdateInfo
 	checkpointFile := fmt.Sprintf("%s.checkpoint.json", pathPrefix)
 	return b.bucket.Copy(context.TODO(), checkpointFile, b.stackPath(name), nil)
 }
+
+// This was copied from
+// https://github.com/pkg/errors/blob/5dd12d0cfe7f152f80558d591504ce685299311e/errors.go#L264-L288.
+func legacyErrorCause(err error) error {
```

Comment:
Is this actually necessary?

I haven't actually tried it out, so it would be good to confirm, but it looks like this is only used for errors returned from `listBucket`, and we're wrapping the error in on place in `listBucket`:

https://github.com/pulumi/pulumi/blob/070757c7a28dff940697ffc0f76f7d01361cd42e/pkg/backend/filestate/bucket.go#L80

I don't believe this wrapped error is going to have a `Cause` method since it's now using `fmt.Errorf` to wrap the original error rather than `errors.Wrap`.

Instead of `legacyErrorCause` shouldn't we be using the built-in [`errors.Unwrap`](https://pkg.go.dev/errors#Unwrap) to get the underlying error (taking care to handle the case where it isn't wrapped)?

Reply:
It doesn't seem to be necessary. I went as far down he call stack as I could (until interfaces). I think it's fine if tests pass.